dashboard-frontend/src/pages/AccountsPage/AccountList.tsx

```diff
@@ -0,0 +1,175 @@
+import PrimaryButton from '../../components/PrimaryButton';
+import InputTypeformKey from './InputTypeformKey';
+import InputFlyAPIKey from './InputFlyAPIKey';
+import useCreateAccount from './useCreateAccount';
+
+const AccountList: React.FC<any> = ({
+  accounts,
+  clearCreateAccounts,
+}: {
+  accounts: AccountResource[];
+  clearCreateAccounts: () => void;
+}) => {
+  return (
+    <ListLayout>
+      {accounts.map((account, index) => (
+        <AccountListItem
+          key={account.authType}
+          account={account}
+          index={index}
+          credentials={account.connectedAccount?.credentials}
+          clearCreateAccounts={clearCreateAccounts}
+        />
+      ))}
+    </ListLayout>
+  );
+};
+
+const AccountListItem: React.FC<any> = ({
+  account,
+  index,
+  credentials,
+  clearCreateAccounts,
+}: {
+  account: AccountResource;
+  index: number;
+  credentials: any;
+  clearCreateAccounts: () => void;
+}) => {
+  const { isCreating, errorOnCreate, createAccount } = useCreateAccount();
+  const [credential, setCredential] = useState(credentials);
+
+  // used to handle the state of the credentials attached
+  // to the account
+  function handleCredentialChange(e: any): void {
+    setCredential({
+      ...credential,
+      [e.target.name]: e.target.value,
+    });
+  }
+
+  // a bit of duck typing to return the correct input box
+  // TODO this seems complicated, we shouldnt need it
+  const inputType: any = {
+    typeform: (
+      <InputTypeformKey
+        error={errorOnCreate}
+        handleCredentialChange={handleCredentialChange}
+        entity={account.authType}
+        index={index}
+        typeformkey={credential.key}
+      />
+    ),
+    fly: (
+      <InputFlyAPIKey
+        error={errorOnCreate}
+        handleCredentialChange={handleCredentialChange}
+        index={index}
+        entity={account.authType}
+        flyApiKey={credential.api_key}
+      />
+    ),
+  };
+
+  const validateCredentials = JSON.parse(
+    JSON.stringify(credential),
+    (key, value) => (value === null || value === '' ? undefined : value)
+  );
+
+  const handleSubmitForm = (e: any): void => {
+    e.preventDefault();
+
+    const data = {
+      name: account.name,
+      authType: account.authType,
+      connectedAccount: {
+        createdAt: Date.now(),
+        credentials: validateCredentials,
+      },
+    };
+    clearCreateAccounts();
+    // We only ever create accounts as the endpoint is
+    // idempotent
+    createAccount(data);
+  };
+
+  return (
+    <li data-testid="account-list-item">
+      <div className="px-4 py-4 sm:px-6 py-6">
+        <div className="flex flex-col sm:grid sm:grid-cols-5 sm:gap-4">
+          <h2 className="mb-4 text-sm font-medium text-indigo-600 truncate sm:mb-0 sm:col-span-1">
+            {account.name}
+          </h2>
+          <form onSubmit={handleSubmitForm} className="col-span-3">
+            {inputType[account.name] || undefined}
```

Comment:
Same as above as it seems we're only checking for existence rather than intentionally wanting to return undefined so what about this: 

```
inputType[account.name] && {
 <div className="flex items-baseline">
              <PrimaryButton
                type="submit"
                testId={`existing-account-submit-button-${index}`}
                loading={isCreating}
              >
                {account.connectedAccount &&
                account.connectedAccount.credentials
                  ? 'Update'
                  : 'Connect'}
              </PrimaryButton>
            </div>
          </form>
        </div>
      </div>
}
``` 

This is basically saying, if this thing exists, render this, otherwise do nothing.

Reply:
Not sure if im missing something but the above doesnt seem to work, it only shows the button and not the input